src/potentials.h

```diff
@@ -81,46 +190,62 @@ struct Dummy : public PairPotentialBase {
     void to_json(json &) const override;
 }; //!< A dummy pair potential that always returns zero
 
-struct ParametersTable {
-    enum Mixers { LB, LBSW, HE, NONE };
-    Mixers mixer = NONE;
-    PairMatrix<double> s2, eps; // matrix of sigma_ij^2 and 4*eps_ij (LJ)
-    PairMatrix<double> th, esw; // matrix of squarewell_threshold_ij and squarewell_depth_ij (Square-well)
-    PairMatrix<double> hd, ehe; // matrix of hydrodynamic diameter_ij and energy-strength_ij (Hertz)
-};                              //!< Table of parameters for potential
-
-void from_json(const json &j, ParametersTable &m);
-
-void to_json(json &j, const ParametersTable &m);
 
 /**
- * @brief Lennard-Jones with arbitrary mixing rule
+ * @brief Lennard-Jones potential with an arbitrary combination rule.
  * @note Mixing data is _shared_ upon copying
  */
-class LennardJones : public PairPotentialBase {
+class LennardJones : public MixerPairPotentialBase {
   protected:
-    std::shared_ptr<ParametersTable> m; // table w. sigma_ij^2 and 4xepsilon
+    TPairMatrixPtr sigma_squared; // sigma_ij * sigma_ij
+    TPairMatrixPtr epsilon_quadrupled; // 4 * epsilon_ij
+    void initPairMatrices() override;
 
   public:
-    LennardJones(const std::string &name = "lennardjones");
+    LennardJones(const std::string &name = "lennardjones", const std::string &cite = std::string(),
+                 ECombinationRule combination_rule = COMB_LORENTZ_BERTHELOT)
+        : MixerPairPotentialBase(name, cite, combination_rule) {};
 
     inline Point force(const Particle &a, const Particle &b, double r2, const Point &p) override {
-        double s6 = powi(m->s2(a.id, b.id), 3);
+        double s6 = powi((*sigma_squared)(a.id, b.id), 3);
         double r6 = r2 * r2 * r2;
         double r14 = r6 * r6 * r2;
-        return 6. * m->eps(a.id, b.id) * s6 * (2 * s6 - r6) / r14 * p;
+        return 6. * (*epsilon_quadrupled)(a.id, b.id) * s6 * (2 * s6 - r6) / r14 * p;
```

Comment:
`epsilon_quadrupled` sounds to me like eps^4 (I know now that it's times four.) What about `four_times_epsilon`

Reply:
Powers are based on geometric shapes, i.e., squared, cubed, while multiples are expressed as doubled, tripled, quadrupled, etc. I would keep it.